### ISSUE TITLE
some logging stops working

### ISSUE THREAD

**Issue by miroR** [balabit/syslog-ng#766]
It appears to me related to #121 (which is closed). 
Pls. see:  exec_logging (and more) not working  http://forums.grsecurity.net/viewtopic.php?f=3&t=4308 
where I posted all the details.
As previously (for which see:  Syslog-ng from Delay Logging to BrokenPipe/no Logging https://forums.gentoo.org/viewtopic-t-1001994.html#7838704 )
Reverting to 3.4.8 solved the issue. I just thought you devs should be informed about it.

**Comment by ihrwein** [comment]
Thanks for opening the issue!

#121 was fixed (in 3.6.4) by overwriting the timestamps of the read kernel logs:
```
Kernel messages read from /dev/kmsg may not have accurate timestamps.
The  time  source  used for the logs is not updated after system
 SUSPEND/RESUME. With this patch we ignore the original (and possibly
 inaccurate) timestamp and use the time of reception.
```

It seems like it wasn't forward ported to `3.7.1` or even `3.7.2` ... I just opened #769 and #771.

These PRs will solve the kernel log time drift but this exec_logging should be an other problem.

Could you create a new post on the forum that this patch solves the time drift, please?

https://github.com/balabit/syslog-ng/commit/75dc0c6d4fffeac7a7b0461a6cdc6edd7a62c22c.patch

Could you also paste here a backtrace when syslog-ng is spinning consuming your CPU? Thanks!

**Comment by miroR** [comment]
I was busy for more than a week. And can't easily find time now, if at all.
But why would exec_logging be a problem, I am curious to know?
As for the trace, then I would need to use the GNU debugger, and that may not be easily possible if this issue: has not been solved...  And these wo
GNU debugger checking for PaX and refusing to work with it
https://forums.gentoo.org/viewtopic-t-1011162.html  has not been solved...  And these would need sortingg for me to know and be able to report that all has been fixed on the forums... Calling by to tell, but can't promise...

**Comment by toralf** [comment]
You made it again (3.7.1) after 3.6.4 was fixed (or ?):

tor-relay log # grep SYN *log
kern.log:Nov 20 22:26:29 tor-relay kernel: [2431358.124515] TCP: request_sock_TCP: Possible SYN flooding on port 80. Sending cookies.  Check SNMP counters.
syslog:Nov 20 22:02:34 tor-relay kernel: TCP: request_sock_TCP: Possible SYN flooding on port 80. Sending cookies.  Check SNMP counters.

**Comment by ihrwein** [comment]
@toralf Yeah, somehow those patches were left out :( But now, they are integrated so the timedrift shouldn't appear in the future.

We still need a backtrace when syslog-ng spins. syslog-ng should produce a core file if you send SIGQUIT (kill -3) to it.

**Comment by toralf** [comment]
still - or again - a problem with 3.7.2 : http://www.zwiebeltoralf.de/pub/syslog-ng-3.7.2-mess.txt
at this stable hardened 64 Bit Gentoo, 3.6.4 was fine AFAICT

**Comment by ihrwein** [comment]
@toralf Could you manually apply the `keep-timestamp(no)` option in your configuration? Our patch did exactly this and I'd like to see how it works if it's manually applied. 

Sorry for the inconvenience, maybe this bug is tougher that wn thought.

**Comment by toralf** [comment]
will do it.
BTW completely OT, but one question: In the github email I read "tougher that wn thought."
Did you corrected the typo in the comment above afterwards or does github become scatty  ?

**Comment by ihrwein** [comment]
@toralf I corrected it (and just noticed an other typo... I will keep it there :))

**Comment by toralf** [comment]
fixed in 3.7.3 AFAICT

**Comment by bkil-syslogng** [comment]
Closed successfully as per https://github.com/balabit/syslog-ng/issues/766#issuecomment-206892823 and the linked forum posts
